docs/hacking/coding_best_practices.md

```diff
@@ -0,0 +1,280 @@
+# Some suggestions on how to structure your code to pass code review as fast as possible
+
+## Table of contents
+
+* [Background](#background)
+* [Best practices](#best-practices)
+  * [Document your decisions](#document-your-decisions)
+  * [Write easy to follow code](#write-easy-to-follow-code)
+  * [Favor styles which are easier to reason about](#favor-styles-which-are-easier-to-reason-about)
+  * [Write clear tests which actually test functionality](#write-clear-tests-which-actually-test-functionality)
+ * [FAQ](#faq)
+
+## Background
```

Comment:
Altough I think this section is very well written and is pertinent, I'm afraid that the large text might scare off some people from reading. I feel like the two first paragraphs have the most important messages (we enforce good standards for maintainability, and reviews are done in good faith).

In any case, I'm not strongly against keeping this section as is, so please do so if you prefer.

Reply:
I was taking a wider look at both CONTRIBUTING.md and coding_best_practices.md, and they might be a little too much information for a newcomer. Maybe we can add some quick reference summaries later on, so one can get up to speed quickly.

Reply:
This is intended to be more of a supplementary text and not a must-read. All I do here is reiterate basics. I'll reword the referring sentence in CONTRIBUTING.md to highlight this.